src/movableCoord.js

```diff
@@ -514,142 +518,127 @@ eg.module("movableCoord", ["jQuery", eg, window, "Hammer"], function($, ns, glob
 			 * @event
-			 * @param {Object} param
-			 * @param {Number} param.duration
-			 * @param {Array} param.depaPos departure coordinate <ko>현재 좌표</ko>
-			 * @param {Number} param.depaPos.0 departure x-coordinate <ko>현재 x 좌표</ko>
-			 * @param {Number} param.depaPos.1 departure y-coordinate <ko>현재 y 좌표</ko>
-			 * @param {Array} param.destPos destination coordinate <ko>애니메이션에 의해 이동할 좌표</ko>
-			 * @param {Number} param.destPos.0 destination x-coordinate <ko>x 좌표</ko>
-			 * @param {Number} param.destPos.1 destination y-coordinate <ko>y 좌표</ko>
-			 * @param {Boolean} param.isBounce When an animation is bounced, a value is 'true'.  <ko>바운스 되는 애니메이션인 경우 true</ko>
-			 * @param {Boolean} param.isCircular When the area is circular type, a value is 'true'. <ko>순환하여 움직여야하는 애니메이션인경우 true</ko>
-			 * @param {Function} param.done If user control animation, user must call this function. <ko>애니메이션이 끝났다는 것을 알려주는 함수</ko>
-			 * @param {Object} param.hammerEvent Hammerjs event. if you use api, this value is null. http://hammerjs.github.io/api/#hammer.input-event <ko>사용자의 액션에 대한 hammerjs 이벤트 정보 (API에 의해 호출될 경우, null 을 반환)</ko>
-			 *
 			 */
-			var retTrigger = this.trigger("animationStart", animationParam);
+			this.trigger("animationEnd");
+		},
+
+		_animate: function(param, complete) {
+			console.trace(param.destPos, param.duration);
+			param.startTime = new Date().getTime();
+			this._status.animationParam = param;
+			if (param.duration) {
+				var info = this._status.animationParam;
+				var self = this;
+				(function loop() {
+					self._raf = null;
+					if (self._frame(info) >= 1) {
+						// deferred.resolve();
+						complete();
+						return;
+					} // animationEnd
+					self._raf = ns.requestAnimationFrame(loop);
+				})();
+			} else {
+				this._triggerChange(param.destPos, false);
+				complete();
+			}
+		},
 
+		_animateTo: function(absPos, duration) {
+			var param = this._prepareParam(absPos, duration);
+			var retTrigger = this.trigger("animationStart", param);
 			// You can't stop the 'animationStart' event when 'circular' is true.
-			if (isCircular && !retTrigger) {
+			if (param.isCircular && !retTrigger) {
 				throw new Error(
 					"You can't stop the 'animation' event when 'circular' is true."
 				);
 			}
-			animationParam.depaPos = pos;
-			animationParam.startTime = new Date().getTime();
-			this._status.animationParam = animationParam;
+
 			if (retTrigger) {
-				if (animationParam.duration) {
-					// console.error("depaPos", pos, "depaPos",destPos, "duration", duration, "ms");
-					var info = this._status.animationParam;
-					var self = this;
-					(function loop() {
-						self._raf = null;
-						if (self._frame(info) >= 1) {
-							return done(true);
-						} // animationEnd
-						self._raf = ns.requestAnimationFrame(loop);
-					})();
-				} else {
-					this._triggerChange(animationParam.destPos, false);
-					done(!isBounce);
+				var self = this;
+				var queue = [];
+				var flushQueue = function() {
```

Comment:
OK. I will fix it. but, I wish that we can use promise patterns.
If we could use jQuery 1.8+, I can change promise patterns.
anyway, thank you